Framework/API/inc/MantidAPI/Sample.h

```diff
@@ -37,6 +37,18 @@ class MANTID_API_DLL Sample {
   ~Sample();
   Sample &operator=(const Sample &rhs);
 
+  /** @name String representations of geometry flags.*/
+  //@{
+  /// Flag Value 1 = Cylinder
+  constexpr static char *CYLINDER{"Cylinder"};
+  /// Flag Value 2 = FlatPlate
+  constexpr static char *FLAT_PLATE{"Flat plate"};
```

Comment:
I ran the following quick test to check saving out of the different geometry values using the data that Steve provided:

1) I ran the reduction.
2) Keeping the SANS dialog open (minimised), in the Python editor in Mantid I ran the following to change the geometry flag on the workspace to flat plate:

```
ws = mtd['104101_sans_nxs']
ws.sample().setGeometryFlag(2)
```

3) Then I went back to the SANS interface and ran the reduction again (clicked the Process button) and I got an error about `FlatPlate` not being an allowed Geometry property value.

I think the error happens because the enum returned by `sans.common.file_information.convert_to_shape` defines flat plate as `FlatPlate`, so I think we need to use the same here.

I also noticed that the `SampleShape` enum on the Python side doesn't define single crystal, so `convert_to_shape` will return None for a geometry flag of 4, so I think we need to extend the support to cover "SingleCrystal". If you repeat the above test using geometry flag 4 it currently silently saves the geometry as "Disc", which doesn't feel ideal so I wonder if it would be better to be letting the user know there's an issue with the geometry.

Reply:
So for some reason SANSBatchMode uses the `Flat plate` form rather than the `FlatPlate` form, so I wasn't certain of which one to use. 

You're definitely right about doing more to inform the user though. That should probably be a warning at least rather than letting it be silently changed.

Reply:
For `FlatPlate` vs `Flat plate`, I think we need to be using the same value everywhere, as if I'm following correctly than the code in `SANSBatchMode.py` is trying to pass geometry properties to save algorithms as well so I would expect this to produce an error too. If we use the enum everywhere then that might help maintain some consistency.

Reply:
Having said that, looking at the original validator values for the `SaveCanSAS1D` geometry property, I think we want to stick with what was originally in use (and I assume Steve hasn't asked us to start supporting Single Crystal). So on line 168 of `SANSSave.py` I think we need to be assigning the value "Flat plate" rather than the value we get from the enum. Given how confusing this is at the moment, it might be good if we could define the geometry values in a re-usable way on the Python side rather than have string literals in multiple places.

Reply:
I agree. I wonder if we can export the values from Sample.h to python and use them from there. That might be the best way to maintain consistency

Reply:
That would be ideal if it was possible - definitely worth a look, thanks.

Reply:
I had a bit more of a look around with this and it seems that the `FlatPlate` form is more commonly used in the project and `Flat plate` is generally only used by the SANS interface. 

From what Steve has told me, the sample information is intended just to be read rather than processed by any kind of data analysis process. So, as long as it's in a readable format and is consistent I think we can go with the one more commonly used and solidify the format now. 

My ideal would be to have everything from here on be saved as `FlatPlate` but allow our loading algorithms to convert `Flat plate` to the right geometry flag at load time for the sake of backwards compatibility.